source/develop/release-management/features/ovirt-ovn-provider.md

```diff
@@ -0,0 +1,334 @@
+and knowing the physical location of the port update local ovs flows.
+
+#### Detailed steps of nic plugging
+
+* the engine checks if the port for the VM NIC exists, issuing a GET port request to the provider. The provider
+in turn queries the Logical_Switch_Port table of the OVN north DB
+*   if no port exists, the engine will create a new port by issuing a POST port request to the provider. The provider
+    in turn adds a row to the Logical_Switch_Port table of the OVN north DB	
+*   if a port already exists, the engine will update the port by issuing a POST port request to the provider. The provider
+    in turn updates the port row in the Logical_Switch_Port table of the OVN north DB
+*   in both cases the new/updated port is identified by a <PORT ID>, which is the uuid of the port row in the Logical_Switch_Port table of the OVN north DB
+*   the engine will send a nic plug request to VDSM, passing the <PORT ID>
+    as one of the parameters. The <PORT ID> is passed to the VIF driver (VDSM hook) as  a "vnic_id" parameter.
+*   on VDSM, the VIF Driver, invoked using the VDSM before_nic_hotplug/before_device_create hook,
+    will connect the VM NIC to the network provided by the external provider by modifing the device xml
+
+![](external_network_provider_schema1.png "fig:external_network_provider_schema1.jpg")
+
+### Vnic unplug
+
+When a VNIC is unplugged, libvirt will automatically unplug the port from
+the OVS bridge.
+This is the equivalent of executing the following OVS commands:
+	ovs-vsctl del-port <nic name>
+
+This deletes the port from the OVN integration bridge. OVN controller 
+modifies the local OpenFlows on the host and deletes the chassis id from the
+bindings. Other OVN-controllers notice the binding change, and update their
+local flows.
+
+## Packaging and installation
+
+### OVN Central Server
+The OVN central server must be intalled manually. It must be accessible via network to
+the OVN provider and to all the hosts which use OVN.
+
+### OVN Controller
+The OVN Controller must be installed on each host using OVN.
+After installation the OVN controller must be configured with the value of the OVN central
+which it will be using and the local IP used for tunneling.
+This could be automated using an otopi plugin, but this would require a one-time configuration
+of the plugin. A way to establish the tunneling IP on the host should be provided, as well as
+the IP of the OVN central.
+ A look at the plugin directory might be helpful to understand the process:
+/usr/share/ovirt-host-deploy/plugins/ovirt-host-deploy
+
+Another possibility would be to to use vdsm-tool configure.
+
+### Provider packaging and installation
+The provider will be delivered as an RPM. After installation the only configurable
+property will be the IP of the OVN north DB. If not specified, the provider will assume
+it is running on the same host as the provider is installed on.
+
+### Driver packaging
```

Comment:
Yes, but the installation process is separate.
We first have to install OVS and OVN controller, and configure it. Only then do we proceed with installing the driver.
In the end we will probably use the same process to install both together, but logically these are two different configuration items (hence separate description for both).